.github/workflows/nightly_dependency_tests.yaml

```
@@ -0,0 +1,76 @@
+name: Nightly dependencies@develop
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 23 * * *"
+
+concurrency:
+  # github.workflow: name of the workflow, so that we don't cancel other workflows
+  # github.event.pull_request.number || github.ref: pull request number or branch name if not a pull request
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  # Cancel in-progress runs when a new workflow with the same group name is triggered
+  # This avoids workflow runs on both pushes and PRs
+  cancel-in-progress: true
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.12
+
+      - name: Check formatting with pre-commit
+        run: |
+          python -m pip install pre-commit
+          pre-commit run ruff
+
+  Tests:
+    needs: style
```

```suggestion
jobs:
  nightly_tests:
```

I feel we should drop the `style` checks from the nightly tests. If they fail, the nightly tests will fail to run.